clients/fides-js/src/lib/tcf.ts

```diff
@@ -56,6 +84,31 @@ export const generateTcString = async (
         );
         if (consented && vendorIsGvl(vendorPreference)) {
           tcModel.vendorConsents.set(+vendorPreference.id);
+          const thisVendor = experience.tcf_vendors?.filter(
```

Comment:
could we ever have a state where a vendor does not exist on ` experience.tcf_vendors`? Maybe just a console warn if `thisVendor` is null or undefined

Reply:
I don't _think_ we can get into that state. this is kind of a result of trying to piece the `TcfSavePreferences` with the experience. In theory, all those preferences should have come from the experience, so they should all have matches. the alternative was to include more data on the `TcfSavePreferences` payload but that made other parts work less cleanly...

Reply:
yes in theory it should all match up :)